### ISSUE TITLE
Text wrapping cuts words when it shouldn't

### ISSUE THREAD

**Issue by bogcow** [godotengine/godot#88529]
### Tested versions

4.2 stable

### System information

Windows 10

### Issue description

I am using a 3 RichTextLabel nodes, with text wrapping set to Word(Smart).
![GIF 2-18-2024 8-04-17 PM](https://github.com/godotengine/godot/assets/157412753/09e27954-6a9f-4cf2-9c35-16e7d2fd9a5f)
Notice how the wrapping differs in these 3 cases. In the first case, it wraps all the text after the slash. In the second case, it wraps the entire word. In the third case, it wraps everything after the image, including the image, but nothing before.

The expected behavior would be that the entire word wraps without breaking in the middle in all 3 cases, at least from a usability standpoint. In the current state, you cannot have any words with slashes in them, or words with images in them, because the wrapping will cut the words in half, making them display awkwardly. I believe this also happens with some other punctuation marks as well, not just slashes.

### Steps to reproduce

Create a RichTextLabel node, type some words with slashes or images in the middle of them, and notice how the wrapping cuts words.

### Minimal reproduction project (MRP)

N/A

**Comment by Mickeon** [comment]
The second case ("AB.CD") is not a bug. Wrapping after the period would mean things like "...", "Etc.." may not be accounted for properly. Try writing "AB.CD" in this very GitHub textbox and you may notice the same behavior.
 I believe there _are_ some characters that act as separators but are essentially invisible and occupy no space. Those could be used in the middle for these occasions(?).

Most egregiously though, I believe @AThousandShips considers this to be a documentation issue because of the third case. When you insert an image in a **RichTextLabel**, it's not a custom character of sorts and the **RichTextLabel** likely breaks the text this way by design.

**Comment by bogcow** [comment]
For what it's worth, this behavior feels like a bug because it means you can't have a word (e.g., a name of a character or location) with a slash in it, because it will look strange when it wraps. And there isn't really a workaround for it either except just not having slashes, which doesn't feel satisfactory.

**Comment by Mickeon** [comment]
> And there isn't really a workaround for it either except just not having slashes, which doesn't feel satisfactory.

Yes there is. You can use a [word joiner](https://symbl.cc/en/2060/) character. I am aware it sounds unorthodox but it does exist.

"AB/⁠CD". It is written here after the slash but you cannot see it. Or here: `⁠`

**Comment by AThousandShips** [comment]
Testing with other editors for example it *does* break with the slash, but *not* with a period, with a period it strips characters off the end like it would for a complete word

It does it for `-` as well, but does *not* do it for backslash (This is Kate editor)

With notepad it splits on backslash and `-` but *not* on period or forward slash

So it varies
